libc/src/stdio/generic/puts.cpp

```diff
@@ -15,8 +15,21 @@
 
 namespace LIBC_NAMESPACE {
 
+// Simple helper to unlock the file once destroyed.
+struct ScopedLock {
+  ScopedLock(LIBC_NAMESPACE::File *stream) : stream(stream) { stream->lock(); }
+  ~ScopedLock() { stream->unlock(); }
+
+private:
+  LIBC_NAMESPACE::File *stream;
+};
+
 LLVM_LIBC_FUNCTION(int, puts, (const char *__restrict str)) {
```

Comment:
why this is marked as `__restrict`?

Reply:
I think all of the implementations of string functions are like this. My best guess is that it's giving the same effect as `const char const *` without changing the signature or something.

Reply:
Sorry, can you please provide more info on this point; it doesn't sound right to me.

My understanding of `__restrict` is that it has to do with strict aliasing of function parameters. If there's only 1 parameter, there is __nothing__ to be concerned with wrt to aliasing.

So why add __restrict here?